### PR TITLE
Refs #35058 -- Added is_3d and set_3d() to OGRGeometry.

### PR DESCRIPTION
**Pull request by smithdc1**

### CODE REVIEW

File: django/contrib/gis/gdal/geometries.py [hunk 1]
```diff
@@ -268,6 +268,20 @@ def extent(self):
...
+        return capi.is_3d(self.ptr)
+
+    def set_3d(self, value):
+        """Set if this geometry has Z coordinates."""
+        if value is True:
+            capi.set_3d(self.ptr, 1)
+        elif value is False:
+            capi.set_3d(self.ptr, 0)
+        else:
+            raise ValueError(f"Input to 'set_3d' must be a boolean, got {value!r}.")
```
Comment (felixxm) [inline django/contrib/gis/gdal/geometries.py:283]: Can we also switch to these methods when possible? e.g. `OGRGeometry.z`, `OGRGeometry.tuple`, or `verify_geom()` as you did in #17696.

File: docs/ref/contrib/gis/gdal.txt [hunk 2]
```diff
@@ -556,6 +556,27 @@ coordinate transformation:
...
+    .. versionadded:: 5.1
+
+    A method to add or remove the Z coordinate dimension.
+
+    .. code-block:: pycon
+
+        >>> p = OGRGeometry("POINT (1 2 3)")
+        >>> p.is_3d
+        True
+        >>> p.set_3d = False
```
Comment (felixxm) [inline docs/ref/contrib/gis/gdal.txt:576]: `set_3d()` is a method
```suggestion
        >>> p.set_3d(False)
```

File: tests/gis_tests/gdal_tests/test_geom.py [hunk 3]
```diff
@@ -714,3 +714,13 @@ def test_geometry_types(self):
...
+
+    def test_is_3d_and_set_3d(self):
+        geom = OGRGeometry("POINT (1 2)")
+        self.assertEqual(geom.is_3d, False)
+        geom.set_3d(True)
+        self.assertEqual(geom.is_3d, True)
+        self.assertEqual(geom.wkt, "POINT (1 2 0)")
+        geom.set_3d(False)
+        self.assertEqual(geom.is_3d, False)
+        self.assertEqual(geom.wkt, "POINT (1 2)")
```
Comment (claudep) [inline tests/gis_tests/gdal_tests/test_geom.py:726]: Maybe add a line to test the `ValueError` where `set_3d` is use with non-boolean value?

Comment (felixxm) [reply tests/gis_tests/gdal_tests/test_geom.py:726]: Yes, definitely :dart:

File: docs/releases/5.1.txt [hunk 4]
```diff
@@ -71,6 +71,9 @@ Minor features
   for backward compatibility.
 
 * :class:`~django.contrib.gis.measure.Area` now supports the ``ha`` unit.
+* The presence ``Z`` dimension for geometries can now be checked by
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/releases/5.1.txt:74]: We can break it down into 2 points:
```
* The new :attr:`.OGRGeometry.is_3d` attribute ...

* The new :meth:`.OGRGeometry.set_3d` method ...
```

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: 

I pushed tiny edits.

**Pull request merged**